### PR TITLE
Explore organizations

### PR DESCRIPTION
**Pull request by poonam-ghewande**
### Summary 

1. Added explore organization, details with available credential and connection
2. Update loader in application
3. Update connection flow

### Issue ticket number and link    
add ticket link

### Checklist before requesting a review
- [x]  All commits contain a DCO Signed-off-by line (we use the DCO GitHub app to enforce this).
- [x]  Self-review of code & No unnecessary commented-out code.
- [x]  All existing functionality remain intact.
- [x]  Updated the README if necessary..
- [x]  No hard-coded sensitive information.

### CODE REVIEW

File: app/components/common/AlphabetListView.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+const initState = {
+  selectAlphabet: '',
+  itemHeight: 0,
+}
+
+type State = Readonly<typeof initState>
+
+class AlphabetListView extends PureComponent<IProps, State> {
```
Comment (sairanjit) [inline app/components/common/AlphabetListView.tsx:23]: Convert this component to functional component

File: app/components/common/ScanButton.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { useNavigation } from '@react-navigation/core'
+import React from 'react'
+import { View, StyleSheet, TouchableOpacity, Image } from 'react-native'
+
+import { Screens } from '../../types/navigators'
+
+const ScanButton: React.FC<Props> = () => {
```
Comment (sairanjit) [inline app/components/common/ScanButton.tsx:7]: `Props` where is the props values used from ?

File: app/components/common/SectionHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import { FC } from 'react'
+import { View } from 'react-native'
+
+export interface IProps {
+  title: string
+}
+
+export const sectionHeaderHeight = 10
+
+const SectionHeader: FC<IProps> = function () {
```
Comment (sairanjit) [inline app/components/common/SectionHeader.tsx:11]:
```suggestion
const SectionHeader: FC<IProps> = () => {
```suggestion
const SectionHeader: FC<IProps> = function () {
```
```

File: app/components/common/SectionListItem.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+const SectionListItem: FC<IProps> = function (props) {
+  return (
+    <View style={[styles.sectionListItemContainer, { height: props.height }]}>
+      <View
+        style={[
+          styles.sectionListItemWrapper,
+          {
+            backgroundColor: props.active ? '#1F4EAD' : 'transparent',
```
Comment (sairanjit) [inline app/components/common/SectionListItem.tsx:20]: Move colors to theme

File: app/components/common/SectionListItem.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+import styles from './styles'
+
+interface IProps {
+  title: string
+  height: number
+  active: boolean
+}
+
+const SectionListItem: FC<IProps> = function (props) {
```
Comment (sairanjit) [inline app/components/common/SectionListItem.tsx:13]: Use arrow function syntax

File: app/components/common/SectionListItem.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+          styles.sectionListItemWrapper,
+          {
+            backgroundColor: props.active ? '#1F4EAD' : 'transparent',
+          },
+        ]}>
+        <Text
+          style={[
+            styles.sectionListItemText,
+            {
+              color: props.active ? 'white' : '#333',
```
Comment (sairanjit) [inline app/components/common/SectionListItem.tsx:27]: Use colors from theme

File: app/components/common/styles.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import { StyleSheet } from 'react-native'
+
+export default StyleSheet.create({
+  container: {
+    flex: 1,
+  },
+  sectionHeaderContainer: {
+    justifyContent: 'center',
+    backgroundColor: '#F4F4F4',
```
Comment (sairanjit) [inline app/components/common/styles.ts:9]: Use colors from theme

File: app/components/common/styles.ts [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    width: 16,
+    height: 16,
+    borderRadius: 8,
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  sectionListItemText: {
+    fontSize: 10,
+    textAlign: 'center',
+    fontFamily: 'PingFangSC-Regular',
```
Comment (sairanjit) [inline app/components/common/styles.ts:32]: Use font family from theme

File: app/defaultConfiguration.ts [hunk 9]
```diff
@@ -1,5 +1,5 @@
 import { BrandingOverlayType, RemoteOCABundleResolver } from '@hyperledger/aries-oca/build/legacy'
-import Config from 'react-native-config'
+import { Config } from 'react-native-config'
```
Comment (sairanjit) [inline app/defaultConfiguration.ts:2]: Is it a default import?

File: app/navigators/CredentialStack.tsx [hunk 10]
```diff
@@ -15,6 +15,7 @@ import { createDefaultStackOptions } from './defaultStackOptions'
 const CredentialStack: React.FC = () => {
   const Stack = createStackNavigator<CredentialStackParams>()
   const theme = useTheme()
+  const { scan } = useConfiguration()
```
Comment (sairanjit) [inline app/navigators/CredentialStack.tsx:18]: Use `Scan` component directly from screens

File: app/navigators/HomeStack.tsx [hunk 11]
```diff
@@ -34,6 +36,7 @@ const HomeStack: React.FC = () => {
           title: t('Screens.Notifications'),
         })}
       />
+      <Stack.Screen name={Screens.Scan} component={scan} />
```
Comment (sairanjit) [inline app/navigators/HomeStack.tsx:39]: Use `Scan` component directly from screens

File: app/localization/en/index.ts [hunk 12]
```diff
@@ -550,8 +550,9 @@ const translation = {
     "Success"  : 'Success',
     "CredentialDetailsW3C" :'Credential Details W3C',
     "WalletName" :"Wallet Name",
-    "ProofChangeCredential": "Choose a credential",
     "Organization": "Explore",
```
Comment (sairanjit) [inline app/localization/en/index.ts:553]: Please use proper key values for localization `"Organization": "Explore"`

File: app/navigators/TabStack.tsx [hunk 13]
```diff
@@ -105,77 +103,8 @@ const TabStack: React.FC = () => {
...
-                    </Text>
-                  </View>
-                </AttachTourStep>
-              </View>
-            ),
-            tabBarShowLabel: false,
-            tabBarAccessibilityLabel: t('TabStack.Scan'),
-            tabBarTestID: testIdWithKey(t('TabStack.Scan')),
-          }}
-          listeners={({ navigation }) => ({
```
Comment (sairanjit) [inline app/navigators/TabStack.tsx:167]: Please verify if we use `assertConnectedNetwork` while using scan screen beacuse that helps us in restricting user to scan qrcodes if device is not connected to internet

File: app/screens/ContactItem.tsx [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+  orgSlug: string
+}
+export const CONTACT_ITEM_HEIGHT = 70
+
+const ContactItem: FC<IItemProps<IContact>> = function ({ item, last }) {
+  return (
+    <View key={item.id} style={[{ borderBottomWidth: last ? 0 : StyleSheet.hairlineWidth }]}>
+      <Image source={{ uri: item.logoUrl }} />
+      <View>
+        <Text>{`${item.name}`}</Text>
```
Comment (sairanjit) [inline app/screens/ContactItem.tsx:20]: Use colors from theme

File: app/screens/ListCredentials.tsx [hunk 15]
```diff
@@ -108,6 +109,9 @@ const ListCredentials: React.FC = () => {
         ListEmptyComponent={() => <CredentialEmptyList message={t('Credentials.EmptyList')} />}
       />
       <CredentialListOptions />
+      <View style={{ justifyContent: 'flex-end', marginTop: '90%' }}>
```
Comment (sairanjit) [inline app/screens/ListCredentials.tsx:112]: Avoid inline styles

File: app/screens/OrganizationDetails.tsx [hunk 16]
```diff
@@ -0,0 +1,305 @@
...
+import { Screens, Stacks } from '../types/navigators'
+import { fetchOrganizationDetail } from '../utils/Organization'
+import { useAppAgent } from '../utils/agent'
+import { connectFromInvitation, getJson, getUrl, receiveMessageFromUrlRedirect } from '../utils/helpers'
+import { testIdWithKey } from '../utils/testable'
+
+interface OrgnizationDetailsProps {
+  name: string
+  description: string
+  logoUrl: any
```
Comment (sairanjit) [inline app/screens/OrganizationDetails.tsx:20]: LogoUrl is string here

File: app/screens/OrganizationList.tsx [hunk 17]
```diff
@@ -1,140 +1,101 @@
-import { useNavigation } from '@react-navigation/core'
-import React, { useState } from 'react'
-import { View, StyleSheet, Text, TextInput, FlatList, TouchableOpacity } from 'react-native'
+import { StackNavigationProp } from '@react-navigation/stack'
+import React, { Fragment, useEffect, useState } from 'react'
+import { View, Text, TextInput, Platform, Image, ActivityIndicator } from 'react-native'
+import { ScaledSheet } from 'react-native-size-matters'
```
Comment (sairanjit) [inline app/screens/OrganizationList.tsx:4]: Please use `react-native-responsive-screen`

File: app/screens/OrganizationList.tsx [hunk 18]
```diff
@@ -144,67 +105,110 @@ const OrganizationList: React.FC = () => {
     orgLabelText: {
       color: ColorPallet.brand.primary,
     },
+    alphabetView: {
+      marginLeft: 20,
+      borderWidth: 1,
+      height: '100%',
+      borderRadius: 5,
+      borderColor: '#E1EAFF',
```
Comment (sairanjit) [inline app/screens/OrganizationList.tsx:113]: Use colors from theme

File: app/screens/OrganizationList.tsx [hunk 19]
```diff
@@ -144,67 +105,110 @@ const OrganizationList: React.FC = () => {
...
+    },
+    highlightedText: {
+      color: 'white',
+    },
   })
-  const filteredOrganizations = organizations.filter(org => {
-    const firstLetter = org.name.charAt(0).toUpperCase()
-    if (selectedLetter && firstLetter !== selectedLetter) {
-      return false
+  const fetchData = async () => {
```
Comment (sairanjit) [inline app/screens/OrganizationList.tsx:133]: convert this to hooks

File: app/screens/OrganizationList.tsx [hunk 20]
```diff
@@ -144,67 +105,110 @@ const OrganizationList: React.FC = () => {
...
+    } finally {
+      setLoading(false)
     }
+  }
+
+  useEffect(() => {
+    fetchData()
+  }, [])
+
+  const filteredOrganizations = orgnizationdata.organizations.filter(org => {
```
Comment (sairanjit) [inline app/screens/OrganizationList.tsx:151]: `orgnizationdata` please check spelling and prefer using camelCase

File: app/screens/OrganizationList.tsx [hunk 21]
```diff
@@ -144,67 +105,110 @@ const OrganizationList: React.FC = () => {
...
+    const initialLetter = String.fromCharCode(letter)
+    data[initialLetter] = items.filter(item => item.name.charAt(0) === initialLetter)
+  }
+  const HEADER_HEIGHT = 50
 
   return (
     <View style={styles.container}>
       <View style={styles.headerTextView}>
-        <Text style={styles.titleText}>Select an organization to discover what credentials you can obtain.</Text>
+        <Text style={styles.titleText}>Select an organization to get credentials</Text>
```
Comment (sairanjit) [inline app/screens/OrganizationList.tsx:180]: Add text to localizations

File: app/utils/Organization.ts [hunk 22]
```diff
@@ -0,0 +1,31 @@
+import Toast from 'react-native-toast-message'
+
+import { ToastType } from '../components/toast/BaseToast'
+
+export const fetchOrganizationData = async () => {
+  try {
+    const response = await fetch('https://devapi.credebl.id/orgs/public-profile')
```
Comment (sairanjit) [inline app/utils/Organization.ts:7]: Add urls to `.env`

File: package.json [hunk 23]
```diff
@@ -75,6 +75,7 @@
     "react-native-safe-area-context": "^4.7.1",
     "react-native-scalable-image": "^1.1.0",
     "react-native-screens": "^3.23.0",
+    "react-native-size-matters": "^0.4.2",
```
Comment (sairanjit) [inline package.json:78]: Please use this package `react-native-responsive-screen`

**Review by sairanjit — CHANGES_REQUESTED** [review]

File: app/components/common/AlphabetFlatList.tsx [hunk 24]
```diff
@@ -0,0 +1,180 @@
...
+  initialNumToRender: number
+  pageY: number
+}
+
+const windowHeight = Dimensions.get('window').height
+function AlphabetFlatList<ItemT>(props: IProps<ItemT>) {
+  const [containerHeight, setContainerHeight] = useState(windowHeight)
+  const [itemLayout, setItemLayout] = useState([])
+  const [titles, setTitles] = useState([])
+  // const [selectAlphabet, setSelectAlphabet] = useState('')
```
[the remark this answers is not in the document]
Comment (poonam-ghewande) [reply app/components/common/AlphabetFlatList.tsx]: removed

File: app/components/common/AlphabetFlatList.tsx [hunk 25]
```diff
@@ -0,0 +1,180 @@
...
+
+  const getItemLayout = (data: any, index: number) => ({
+    length: itemLayout[index].length,
+    offset: itemLayout[index].offset,
+    index,
+  })
+
+  const renderItem = (info: ListRenderItemInfo<string>) => (
+    <View key={info.index}>
+      {/* {props.renderSectionHeader && props.renderSectionHeader({ title: info.item })} */}
```
[the remark this answers is not in the document]
Comment (poonam-ghewande) [reply app/components/common/AlphabetFlatList.tsx]: removed

File: app/screens/OrganizationDetails.tsx (hunk 16 above)
Comment (poonam-ghewande) [reply app/screens/OrganizationDetails.tsx]: added string

File: app/screens/OrganizationList.tsx (hunk 18 above)
Comment (poonam-ghewande) [reply app/screens/OrganizationList.tsx]: added  color in theme

File: app/screens/OrganizationList.tsx (hunk 21 above)
Comment (poonam-ghewande) [reply app/screens/OrganizationList.tsx]: Added to localizations

File: app/screens/OrganizationList.tsx (hunk 20 above)
Comment (poonam-ghewande) [reply app/screens/OrganizationList.tsx]: done

File: app/components/common/AlphabetFlatList.tsx [hunk 26]
```diff
@@ -0,0 +1,180 @@
...
+    })
+
+    let initialNumToRender = itemLayout.findIndex(item => item.offset >= containerHeight)
+    if (initialNumToRender < 0) {
+      initialNumToRender = titles.length
+    }
+
+    setItemLayout(itemLayout)
+    setTitles(titles)
+    // setSelectAlphabet(titles[0])
```
[the remark this answers is not in the document]
Comment (poonam-ghewande) [reply app/components/common/AlphabetFlatList.tsx]: done

File: app/components/common/AlphabetFlatList.tsx (hunk 25 above)
[the remark this answers is not in the document]
Comment (poonam-ghewande) [reply app/components/common/AlphabetFlatList.tsx]: done

File: app/components/common/ScanButton.tsx (hunk 2 above)
Comment (poonam-ghewande) [reply app/components/common/ScanButton.tsx]: removed it was unused.

File: app/components/common/SectionListItem.tsx (hunk 5 above)
Comment (poonam-ghewande) [reply app/components/common/SectionListItem.tsx]: done

File: app/components/common/SectionListItem.tsx (hunk 4 above)
Comment (poonam-ghewande) [reply app/components/common/SectionListItem.tsx]: done

File: app/components/common/SectionListItem.tsx (hunk 6 above)
Comment (poonam-ghewande) [reply app/components/common/SectionListItem.tsx]: done

File: app/components/common/styles.ts (hunk 8 above)
Comment (poonam-ghewande) [reply app/components/common/styles.ts]: done

File: app/localization/en/index.ts (hunk 12 above)
Comment (poonam-ghewande) [reply app/localization/en/index.ts]: done

File: app/navigators/CredentialStack.tsx (hunk 10 above)
Comment (poonam-ghewande) [reply app/navigators/CredentialStack.tsx]: done

File: app/navigators/TabStack.tsx (hunk 13 above)
Comment (poonam-ghewande) [reply app/navigators/TabStack.tsx:167]: verified and done

File: app/screens/ContactItem.tsx (hunk 14 above)
Comment (poonam-ghewande) [reply app/screens/ContactItem.tsx:20]: done

File: app/screens/OrganizationDetails.tsx (hunk 16 above)
Comment (poonam-ghewande) [reply app/screens/OrganizationDetails.tsx]: done

File: app/screens/OrganizationList.tsx (hunk 17 above)
Comment (poonam-ghewande) [reply app/screens/OrganizationList.tsx]: done

File: app/screens/OrganizationList.tsx (hunk 19 above)
Comment (poonam-ghewande) [reply app/screens/OrganizationList.tsx]: done

File: app/utils/Organization.ts (hunk 22 above)
Comment (poonam-ghewande) [reply app/utils/Organization.ts]: done

File: package.json (hunk 23 above)
Comment (poonam-ghewande) [reply package.json]: done

File: app/components/common/AlphabetListView.tsx (hunk 1 above)
Comment (poonam-ghewande) [reply app/components/common/AlphabetListView.tsx]: done

File: app/components/common/SectionHeader.tsx (hunk 3 above)
Comment (poonam-ghewande) [reply app/components/common/SectionHeader.tsx]: done

**Review by sairanjit — APPROVED** [review]

**Pull request merged**